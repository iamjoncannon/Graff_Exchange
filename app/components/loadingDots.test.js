import React from 'react';
import renderer from 'react-test-renderer';
import LoadingDots from './LoadingDots'

describe("loadingDots", ()=>{

    let component 

    beforeEach(()=>{

        component = renderer.create(<LoadingDots size={"3rem"}/>)
    })

    it('1) renders successfully', ()=>{

        let tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
    })

    it("2) renders a loading dots spans and three spans", ()=>{

        const dots = component.root.findAllByType("span")

        expect(dots.length).toEqual(3)
    })

    it("3) sets fontSize style of divs to size prop",()=>{

        const dots = component.root.findAllByType("span")

        expect(dots[0].props.style.fontSize).toEqual("3rem")
    })
})