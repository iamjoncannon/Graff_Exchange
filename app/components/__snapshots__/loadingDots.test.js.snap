// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadingDots 1) renders successfully 1`] = `
<div
  className="loading-dots"
>
  <span
    style={
      Object {
        "color": "black",
        "display": "inline",
        "fontSize": "3rem",
      }
    }
  >
    .
  </span>
  <span
    style={
      Object {
        "color": "black",
        "display": "inline",
        "fontSize": "3rem",
      }
    }
  >
    .
  </span>
  <span
    style={
      Object {
        "color": "black",
        "display": "inline",
        "fontSize": "3rem",
      }
    }
  >
    .
  </span>
</div>
`;
